auction-server/src/auction/service/verification.rs

```diff
@@ -463,42 +519,95 @@ impl Service<Svm> {
         &self,
         transaction: VersionedTransaction,
     ) -> Result<BidDataSvm, RestError> {
-        let submit_bid_instruction = self.verify_submit_bid_instruction(transaction.clone())?;
-        let submit_bid_data = Self::extract_submit_bid_data(&submit_bid_instruction)?;
-
-        let permission_account = self
-            .extract_account(
-                &transaction,
-                &submit_bid_instruction,
-                self.config
-                    .chain_config
-                    .express_relay
-                    .permission_account_position,
-            )
-            .await?;
-        let router = self
-            .extract_account(
-                &transaction,
-                &submit_bid_instruction,
-                self.config
-                    .chain_config
-                    .express_relay
-                    .router_account_position,
-            )
-            .await?;
-        Ok(BidDataSvm {
-            amount: submit_bid_data.bid_amount,
-            permission_account,
-            router,
-            deadline: OffsetDateTime::from_unix_timestamp(submit_bid_data.deadline).map_err(
-                |e| {
-                    RestError::BadParameters(format!(
-                        "Invalid deadline: {:?} {:?}",
-                        submit_bid_data.deadline, e
-                    ))
-                },
-            )?,
-        })
+        let submit_bid_instruction_result = self.verify_submit_bid_instruction(transaction.clone());
+        let swap_instruction_result = self.verify_swap_instruction(transaction.clone());
+
+        assert!(
+            submit_bid_instruction_result.is_err() != swap_instruction_result.is_err(),
+            "Either submit_bid or swap must be present, but not both"
+        );
+
+        if submit_bid_instruction_result.is_ok() {
```

Comment:
I think we can use match here:

match ((submit_bid_instruction_result, swap_instruction_result)) {
    Err(_), Ok(a) => {},
   Ok(a), Err(_) => {},
   _ => anyhow!(...)
}